mule-maven-plugin/src/main/java/org/mule/tools/maven/mojo/GenerateSourcesMojo.java

```diff
@@ -40,10 +42,17 @@
     requiresDependencyResolution = ResolutionScope.RUNTIME)
 public class GenerateSourcesMojo extends AbstractMuleMojo {
 
-
+    private static final String MULE_MODULE_GROUP_ID = "org.mule.modules";
     protected MulePluginResolver mulePluginResolver;
     protected ApplicationDependencySelector applicationDependencySelector;
+    private static Set<String> containerPlugins = new HashSet<>();
 
+    static {
```

Comment:
Please remove this static block 

Either place this logic in the consctructor OR create a pluginFilter Class  with the filtering logic which you can unit test. 

As a general rule static blocks are not a very good practice